core/src/main/java/org/elasticsearch/index/query/ConstantScoreQueryBuilder.java

```diff
@@ -49,7 +54,7 @@
      * @param filterBuilder The query to wrap in a constant score query
      */
     public ConstantScoreQueryBuilder(QueryBuilder filterBuilder) {
-        this.filterBuilder = filterBuilder;
+        this.filterBuilder = Objects.requireNonNull(filterBuilder);
```

Comment:
I am still on the fence. I don't like null values, but I think there is a little difference between the empty query and something that was not set. That is why I don't think the default should be empty (although that might seem very convenient), which is something that requires users to put something (although incomplete) in the json. But with null values null checks are required, and relying on validate seems to be ok but not 100% correct because the java api exposes a lot of public methods that potentially throw NPE if they had no null checks, when executed against a non validated query. This last problem is not huge but it bugs me. In the end I think your last proposal around making mandatory queries required and have null checks in constructor is our best bet. We really want to remove null checks and make sure that those queries are always non null and that is the only way. But it's a breaking change for the java api, which reminds me that we might have already broken bw comp with similar changes without realising it :)